cli/src/main/kotlin/commands/EvaluatorCommand.kt

```diff
@@ -104,6 +114,25 @@ object EvaluatorCommand : CommandWithHelp() {
 
         val script = preface + (rulesFile?.readText() ?: javaClass.getResource(rulesResource).readText()) + postface
 
+        if (syntaxCheck) {
+            if (engine is Compilable) {
+                val canCompile = try {
+                    engine.compile(script)
+                    true
+                } catch (e: ScriptException) {
+                    log.error {
+                        e.message?.let { it } ?: "No error message available."
+                    }
+
+                    false
+                }
+
+                return if (canCompile) 0 else 1
+            } else {
+                throw IllegalArgumentException("The scripting engine does not support compilation.")
```

Comment:
Can you use `require(engine is Compilable)` instead? This would remove one level of indentation and move the error message close to the check.

Reply:
Good idea, done.